src/clj/puppetlabs/puppetserver/certificate_authority.clj

```diff
@@ -1069,21 +1076,21 @@
        :msg (i18n/tru "CSR contains a public key that does not correspond to the signing key")})))
 
 (schema/defn ensure-no-authorization-extensions!
-  "Throws an exception if the CSR contains authorization exceptions. These
-  extensions need to be signed using the puppet cert tool. This may change in
-  the future, but for now, it's too risky that certificates with authentication
-  extensions could be signed unintentionally."
-  [csr :- CertificateRequest]
+  "Throws an exception if the CSR contains authorization exceptions AND the user
+   has chosen to disallow authorization-extensions.  This ensures that
+   certificates with authentication extensions can only be signed intentionally."
+  [csr :- CertificateRequest
+   allow-authorization-extensions :- schema/Bool]
   (let [extensions (utils/get-extensions csr)]
     (doseq [extension extensions]
       (when (utils/subtree-of? ppAuthCertExt (:oid extension))
-        (sling/throw+
-         {:kind :disallowed-extension
-          :msg (format "%s %s"
-                (i18n/trs "CSR ''{0}'' contains an authorization extension, which is disallowed." csr)
-                (i18n/trs "Use {0} to sign this request."
-                     (format "`puppet cert --allow-authorization-extensions sign %s`"
-                             (get-csr-subject csr))))})))))
+        (if (false? allow-authorization-extensions)
+          (sling/throw+
+            {:kind :disallowed-extension
+             :msg (format "%s %s %s"
+                          (i18n/trs "CSR ''{0}'' contains an authorization extension, which is disallowed." (get-csr-subject csr))
+                          (i18n/tru "To allow authorization extensions, set allow-authorization-extensions to true in your puppetserver.conf file,")
```

Comment:
should these also be `trs` rather than `tru`?

Reply:
Doesn't look like we're consistent through the file. I think `trs` respects the server side local and `tru` respects the requesters, ie for things that would be seen in the broswer. Does that sound, right?

Reply:
That sounds fine to me, @Magisus I think you're our i18n guru, what do you think?

Reply:
I actually have no idea how our clojure tooling works, sorry.